File: src/main/java/intro/to/design/patterns/using/javalang/adapter/banner/AdapterBannerMain.java
package intro.to.design.patterns.using.javalang.adapter.banner;

import intro.to.design.patterns.using.javalang.adapter.banner.delegation.PrintDelegation;
import intro.to.design.patterns.using.javalang.adapter.banner.delegation.PrintDelegationBannerDelegation;
import intro.to.design.patterns.using.javalang.adapter.banner.inheritance.PrintInheritance;
import intro.to.design.patterns.using.javalang.adapter.banner.inheritance.PrintInheritanceBannerInheritance;

public class AdapterBannerMain {
    public static void main(String[] args) {
        PrintInheritance printInheritance = new PrintInheritanceBannerInheritance("Hello");
        printInheritance.printWeak();
        printInheritance.printStrong();
        System.out.println();

        PrintDelegation printDelegation = new PrintDelegationBannerDelegation("Hello");
        printDelegation.printWeak();
        printDelegation.printStrong();
        System.out.println();
    }
}

File: typescript-design-pattern/src/into-to-design-patterns-using-java-lang/adapter/banner/adapter-banner-main.ts
import Output from '../../../utils/stdin-stdout/output'
import { Print } from './print'
import PrintBanner from './print-banner'

const adapterBannerMain = () => {
  const print: Print = new PrintBanner('adapter design pattern')
  print.printWeak()
  print.printStrong()
  Output.log()
}

export default adapterBannerMain
